### PR TITLE
Anny fix reports view of team name

### PR DESCRIPTION
**Pull request by astrjc0326**
# Description
image

## Main changes explained:
-Render only 5 teams detail once to speed up the loading time

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. log as admin user
4. go to Other Links → Reports → Reports → Teams
5. open any team on the right in a new tab, check if it render properly
6. select another team name from the list, test if it displays the new selected team

## Screenshots or videos of changes:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/98345990/c0e40d21-cdb7-4930-9c82-edf75c131f5c

## Note:
Although the loading time is faster than before, but it still take some time to load.

### CODE REVIEW

**Review by malikjahanzaib — DISMISSED** [review]
Hi @astrjc0326, I have tested this PR locally and it seems to work as intended. Teams are getting displayed and rendered correctly in both the new tab as well as the original tab, although there is some delay but you have already acknowledged it. Great work!!!

Test rec:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/46221334/5f863a01-d071-47e8-ae23-217a30716d32

**Review by ChirayuRai — CHANGES_REQUESTED** [review]
Hey I checked out your PR and it seems to work on both admin and owner accounts. 

It looks like if you access DIFFERENT teams, it isn't too bad at loading things in. However, when you access the second team while a tab containing the first team's information is already open, fetching the data for the second team takes much longer. 

Also, it seems that opening two (or more) tabs for the SAME team exponentially increases load times for tabs 2 and onward. For instance, I opened [A test team mounica2] and in the first tab, it took anywhere from 1-8 seconds to load in. When I opened the second tab, it took too long to load (over 40 seconds, I gave up on timing it when it hit 40). I think that may need to be investigated, or at the very least there could be a loading spinner added?


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/15188751/512c525b-728c-4e34-b10f-7b92a4ec06bd

**Review by Iris-Song — DISMISSED** [review]
Hi, I have tested this PR locally and it seems to work as intended. Teams are getting displayed and rendered correctly in both the new tab as well as the original tab, although there is some delay but you have already acknowledged it. Good job!

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/58033867/b9b7a46c-b061-450a-9cfe-53f08f92a35c

**Comment by AnishPandita** [comment]
Hey, I tested the PR and the visual changes look good. The new tab for teams opens correctly and all the elements are rendered correctly.
Screenshot 2023-10-28 at 2 38 11 PM
Screenshot 2023-10-28 at 2 38 21 PM

**Comment by kavya151** [comment]
Hello @astrjc0326, I've successfully tested the PR, and it's performing as expected. I opened a new team in a new tab and experimented with opening other teams, both in the same tab and new tabs. However, the previously accessed team consistently reopens. 

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/61841929/75b2df82-87b8-4bed-a962-c16268a8b3ee

**Review by Bali2Mohit — COMMENTED** [review]
Hi @astrjc0326, I tested the PR, and the fix is not working for me. I am still seeing the previous team in both scenarios.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/94716419/8289a7d9-0389-48e4-ac4d-495485b64fc5

**Review by vikrambadhan — COMMENTED** [review]
I've successfully tested the PR, and it's performing as expected. I opened a new team in a new tab and experimented with opening other teams, both in the same tab and new tabs. However, the previously accessed team consistently reopens.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/85417946/e8217671-558e-491a-8de9-07ced9cff8b3

**Review by Shubhu0 — DISMISSED** [review]
Hi @astrjc0326 Tested the Pr and it is working as intended

Uploading HGN APP — Mozilla Firefox 2023-10-21 00-31-12.mp4…

**Comment by palakgosalia** [comment]
Hey, I reviewed your code and tested the changes. It works as intended. Thank you


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/104417272/ae50f8c8-f279-47fa-8828-35ca2cb3d781

**Review by zuhang2000 — COMMENTED** [review]
Hi, I've tested your PR reviews following your instructions. However, I do encounter several errors when I do the testing.
Firstly, the page crashes and pops up a lot of "Unexpected Error" messages. This happens when the page is enduring lots of delays when interacting with.
Secondly, the new team page that I opens in the new tab are not the team that I clicked. I in fact find it so strange that after clicking several teams, this undefined behaviors would happen.
Please find the details in the following recording.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/123051008/6b97d007-201a-4e77-baff-5511d71aaecf

File: src/components/Reports/TeamReport/TeamReport.jsx [hunk 1]
```diff
@@ -167,12 +178,28 @@ export function TeamReport({ match }) {
...
         })
         .then(result => {
+          console.log(result)
           const allTeamMembersPromises = result.map(team => dispatch(getTeamMembers(team._id)));
           Promise.all(allTeamMembersPromises).then(results => {
             setAllTeamsMembers([...results]);
           });
         });
     }
+    return () => {
```
Comment (John-Flavian) [inline src/components/Reports/TeamReport/TeamReport.jsx:188]: Nice job.

The unmount function was definitely neccesary.

**Review by John-Flavian — COMMENTED** [review]
Nice.

**Comment by luisarevalo21** [comment]
Hello @astrjc0326 ,

I tested your PR and loads much faster. Great work! 
However, as others have noted, when clicking on another team, the previous team is loaded instead. 
When i open a new tab the older user is dispalyed as well. 

I also noticed a console.log which displays 200 elements, I don't think its related to your PR but logs shouldn't be rendered on production. 

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/22454953/f05ad9bc-6f99-4546-bbd8-85be8e8b036f

**Comment by Jiyuanxia** [comment]
Hi @astrjc0326 , I have reviewed your pull request and tested it. It seems like there is some issues. When I choose a team and open it on the new tab, it does render me the right page. But after that when I left click a team, it renders me another team page I have ever accesed to.
Screenshot 2023-11-02 141118
Screenshot 2023-11-02 141122
Screenshot 2023-11-02 141353
Screenshot 2023-11-02 141435

**Review by Chuehleo — DISMISSED** [review]
Everything works as expected, great job!

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/122568562/1ab9b100-ab31-458c-b21e-c1262237d142

**Comment by wanglu1207** [comment]
Hi, @astrjc0326. It works as intended! Good job!

**Review by Alforoan — DISMISSED** [review]
It seems to work as intended. Nice!

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/118868270/0c356c6d-6629-4b03-9a52-4c9c70f7d122

**Pull request closed without merging**